src/Fixer/PhpUnit/PhpUnitTestAnnotationFixer.php

```diff
@@ -0,0 +1,375 @@
+            }
+
+            $newFunctionName = $this->addTestToFunctionName($functionName);
+            $tokens[$functionNameIndex] = new Token([T_STRING, $newFunctionName]);
+        }
+    }
+
+    /**
+     * @param string $needle
+     * @param string $haystack
+     *
+     * @return bool
+     */
+    private function startsWith($needle, $haystack)
+    {
+        $len = strlen($needle);
+
+        return substr($haystack, 0, $len) === $needle;
+    }
+
+    /**
+     * @param Tokens $tokens
+     *
+     * @return int[][] array of [start, end] indexes from sooner to later classes
+     */
+    private function findPhpUnitClasses(Tokens $tokens)
+    {
+        $phpUnitTestCaseIndicator = new PhpUnitTestCaseIndicator();
+        $phpunitClasses = [];
+
+        for ($index = 0, $limit = $tokens->count() - 1; $index < $limit; ++$index) {
+            if ($tokens[$index]->isGivenKind(T_CLASS) && $phpUnitTestCaseIndicator->isPhpUnitClass($tokens, $index)) {
+                $index = $tokens->getNextTokenOfKind($index, ['{']);
+                $endIndex = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $index);
+                $phpunitClasses[] = [$index, $endIndex];
+                $index = $endIndex;
+            }
+        }
+
+        return $phpunitClasses;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     *
+     * @return string
+     */
+    private function detectIndent(Tokens $tokens, $index)
+    {
+        if (!$tokens[$index - 1]->isWhitespace()) {
+            return ''; // cannot detect indent
+        }
+
+        $explodedContent = explode("\n", $tokens[$index - 1]->getContent());
+
+        return end($explodedContent);
+    }
+
+    /**
+     * @param string $functionName
+     *
+     * @return string
+     */
+    private function removeTestFromFunctionName($functionName)
+    {
+        if ($this->startsWith('test_', $functionName)) {
+            $newFunctionName = substr($functionName, 5);
+        } else {
+            $functionName = substr($functionName, 4);
+            $newFunctionName = lcfirst($functionName);
+        }
+
+        return $newFunctionName;
+    }
+
+    /**
+     * @param string $functionName
+     *
+     * @return string
+     */
+    private function addTestToFunctionName($functionName)
+    {
+        if (!$this->isCamelCase) {
+            $newFunctionName = 'test_'.$functionName;
+        } else {
+            $newFunctionName = 'test'.ucfirst($functionName);
+        }
+
+        return $newFunctionName;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param $docBlockIndex
+     */
+    private function createDocBlock(Tokens $tokens, $docBlockIndex)
+    {
+        $originalIndent = $this->detectIndent($tokens, $tokens->getNextNonWhitespace($docBlockIndex));
+        $toInsert = [
+            new Token([T_DOC_COMMENT, "/**\n$originalIndent * @test\n$originalIndent */"]),
+            new Token([T_WHITESPACE, "\n".$originalIndent]),
```

Comment:
you need to use whitespace config, and then use proper line ending / indentation.
for now, it will produce wrong code for ppl using tabs / windows line endings.

cover such cases in new test method called `testMessyWhitespaces` (grep for it to get idea what to do)